omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/annotations/Compensable.java

```diff
@@ -72,4 +72,11 @@
    * @return the timeout value
    */
   int timeout() default 0;
+
+  /**
+   * Sending out SagaEnd event to Alpha once the SagaStart annotated method is finished without any error.
+   * Default value is false, which means Omega never send out the SagaEnd event to Alpha once the annotated method is finished.
+   * value is true, means this method is last compensable method need to be called, Omega will send SagaEnd event to Alpha once the method is finished.
+   */
+  boolean sendingSagaEnd() default false;
```

Comment:
javadoc is related to @SagaStart but this class is @Compensable.
Besides, that looks weird to have a method named sendingSagaEnd here.
Would you like to explain the purpose of this method?

Reply:
It just give a chance for the user to send the SagaEnd event out when the local transaction is finished.

Reply:
As to discussion in SCB-1386, it's hard to decide when to send SagaEnd event on the @Compensable method side.

I think it's enough to keep this feature in @SagaStart, no @Compensable

See [comment-1][1], [comment-2][2]

[1]: https://issues.apache.org/jira/browse/SCB-1386?focusedCommentId=16889501&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16889501
[2]: https://issues.apache.org/jira/browse/SCB-1386?focusedCommentId=16889505&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16889505

Reply:
I agree and it could be helpful to introduce the OmegaClient which could be used by the user to control the saga.

Reply:
Current solution just work last transaction is async invocation, if there are more than one async invocations , it's hard to tell when all the local transaction is finished. 
Maybe we can introduce a timeout on Alpha side to verify the finished transactions number, if the all the transactions are committed,  Alpha can mark the Saga transaction is finished.

Reply:
Since Alpha can't know how many transactions a Saga will have in advance, I think it's hard to select a sensible timeout value.

Reply:
The timeout always exists. If you can't evaluate the specific value, you can set the timeout to be smaller than the request timeout. Otherwise, we can dynamically calculate the average request time of the last N and then take the upper limit based on the dynamic distribution ：）

Reply:
Well, the timeout is not the good solution here and it could make the things much more complicated. So at the first, we should make a clear documentation to clarify that we only support the last transaction is async invocation currently. We could have more discuss on the mailing list to get to a better solution for the async invoking. I think perhaps we could introduce the omega client and let the user  call omega.close() explicitly. I know these low client apis should be implemented carefully and avoiding for the abusing.

Reply:
+1 for @zhfeng